### PR TITLE
test: add missing tests to collections.ts for ~100% coverage

### PR DESCRIPTION
**Pull request by is0692vs**
Adds unit tests to `apps/api/src/routes/__tests__/collections.test.ts` to cover error handling paths, unique constraints, and org specific permission logic that were previously uncovered. This brings statement and branch coverage in `collections.ts` up to ~99% / 89% respectively.

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `collections.ts` ルートのテストカバレッジを大幅に拡充し、エラーハンドリング（400/403/404/409/500）、org 権限ロジック、ペーパーフィルタリング、ソート順一括更新など未カバーだったパスを網羅しています。テスト自体の品質面では、2つのテストで無効な visibility 値 `\"member\"` が使われているほか、未使用のモックレスポンスと重複テストがいくつか見られます。
</details>

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

テストのみの変更でプロダクションコードへの影響なし。P2 のみのため安全にマージ可能。

P2 が複数あるが P0/P1 は存在しない。新規テストはいずれも誤った動作を許容するものではなく、カバレッジ向上に実質的に貢献している。

apps/api/src/routes/__tests__/collections.test.ts（visibility "member" の修正、重複テストの整理、余分なモックレスポンスの削除）
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/api/src/routes/__tests__/collections.test.ts | コレクションルートのユニットテストを大幅拡充（185行→648行追加）。エラーハンドリング、権限制御、org コレクションなど多数のパスをカバー。無効な visibility 値（"member"）の使用、未使用モックレスポンス、重複テストの3点が要改善。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[新規テストケース群] --> B{エンドポイント}

    B --> C[POST /collections]
    B --> D[GET /collections/:id]
    B --> E[PATCH /collections/:id]
    B --> F[DELETE /collections/:id]
    B --> G[GET /users/:id/collections]
    B --> H[GET /orgs/:slug/collections]
    B --> I[POST /collections/:id/papers]
    B --> J[DELETE /collections/:id/papers/:paperId]
    B --> K[PATCH /collections/:id/papers]
    B --> L[GET /collections/:id/papers]

    C --> C1[owner_type バリデーション]
    C --> C2[name/slug/description バリデーション]
    C --> C3[visibility バリデーション]
    C --> C4[org オーナー取得 org_slug/owner_id]
    C --> C5[UNIQUE制約エラー → 409]
    C --> C6[一般DBエラー → 500]

    D --> D1[無効トークン → 公開コレクションは200]
    D --> D2[private org → 非adminは404]
    D --> D3[org_only → memberは200]

    E --> E1[404 not found]
    E --> E2[403 forbidden user/org]
    E --> E3[400 invalid JSON/fields]
    E --> E4[409 slug重複]
    E --> E5[500 一般エラー]

    K --> K1[400 missing/extra paper_ids]
    K --> K2[200 batch sort order更新]

    L --> L1[private paper → author/org_onlyでフィルタ]
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 4 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 4
apps/api/src/routes/__tests__/collections.test.ts:151-178
**無効な `visibility` 値を使用している**

2つのテスト（lines 151–163, 165–178）のモックデータに `visibility: "member"` が設定されていますが、プロダクションコードの `VALID_VISIBILITY = ["public", "org_only", "private"]` にこの値は存在しません。本来テストしたいのは `"org_only"` コレクションに対する org メンバーのアクセス制御のはずです。

現状のコードは `canViewCollection` 内で `"public"` でも `"private"` でもない場合のフォールスルーが `isOrgMember(...)` を呼ぶため、偶然にテストが通ります。しかし将来プロダクションコードが有効な visibility 値のみを明示的に処理するよう変更された場合、テストが意図しない方向で壊れる可能性があります。

同様に 165–178 行目の `visibility: "member"` も `"org_only"` に変更してください。

```suggestion
            { getResult: { id: "c1", ownerType: "org", ownerId: "org1", visibility: "org_only" } },
```

### Issue 2 of 4
apps/api/src/routes/__tests__/collections.test.ts:906-919
**既存テストと重複している**

この `"handles invalid JSON body"` テスト（line 906–919）は、842行目に既存の `"POST /api/collections/:id/papers rejects invalid JSON"` テストとまったく同じコードパス（コレクション取得→権限チェック→JSON パース失敗→400）を検証しています。意図的に追加されたものであれば問題ありませんが、カバレッジ向上を意図して追加したなら削除を検討してください。

### Issue 3 of 4
apps/api/src/routes/__tests__/collections.test.ts:1085-1104
**未使用のモックレスポンスがキューされている**

4番目の `{ allResult: [] }` は実際には消費されません。`restrictedRows` には `{ id: "p1", visibility: "private" }` の1件のみが含まれ、`orgOnlyIds` は空（`"private"` は `"org_only"` ではない）となるため、プロダクションコードの `if (orgOnlyIds.length > 0)` ガードにより4回目の `select()` は呼ばれません。この余分なキューはテストの読者を混乱させます。削除して3つのレスポンスに整理してください。

```suggestion
        queueSelectResponses([
            { getResult: { id: "c1", visibility: "public", ownerType: "user", ownerId: "u1" } },
            { allResult: [{ id: "p1", visibility: "private" }, { id: "p2", visibility: "public" }] },
            { allResult: [{ paperId: "p1" }] },
        ]);
```

### Issue 4 of 4
apps/api/src/routes/__tests__/collections.test.ts:1034-1048
**既存テストと重複している**

このテスト（line 1034–1048）は 1393–1429 行目の `"requires every existing paper to be included"` と同一のコードパスを検証しています（`paperIds.length !== existingRows.length` → 400 `"paper_ids must include all papers in collection"`）。削除してもカバレッジへの影響はありません。

`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test: add missing tests to collections.t..."](https://github.com/hiroki-org/openshelf/commit/77bebc5535420dcebfca23d3217e8dee927caf09) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30418618)</sub>

> Greptile also left **4 inline comments** on this PR.

### CODE REVIEW

**Comment by is0692vs** [comment]
対応しました: 無効な visibility 値を org_only に修正、重複テストを削除、未使用のモックレスポンスを削除しました。影響範囲: collections.test.ts。検証: テストが通ることを確認。

**Comment by is0692vs** [comment]
✅ **Fixed all test coverage issues in latest commit (dde06cc):**

1. **Fixed invalid visibility values** (line 151-178): Changed `visibility: "member"` → `visibility: "org_only"`. These are the only valid values: `public`, `org_only`, `private`.

2. **Removed duplicate tests**:
   - Deleted "handles invalid JSON body" test (duplicate of existing "POST /api/collections/:id/papers rejects invalid JSON" at line 842)
   - Deleted "returns 400 when missing existing papers" test (duplicate of "requires every existing paper to be included" at line 1378)

3. **Cleaned up unused mock responses**: Removed 4th unused `{ allResult: [] }` from org_only filter test (since non-org_only papers don't trigger the org check).

All 67 tests pass with improved clarity and no redundancy. ✓

File: apps/api/src/routes/__tests__/collections.test.ts [hunk 1]
```diff
@@ -52,6 +52,185 @@ describe("collections routes", () => {
...
+        ]);
+        const app = await createTestApp();
+        const env = createTestEnv();
+        const res = await app.request("http://localhost/api/collections/c1", {
+            method: "GET",
+            headers: { Authorization: `Bearer ${await createTestJWT({ sub: "user-1" })}` },
+        }, env);
+        expect(res.status).toBe(404);
+        expect(await res.json()).toEqual({ error: "Collection not found" });
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/collections.test.ts]: 対応しました: org collection の visibility は現行差分で有効値の org_only に修正済みです。影響範囲: collections の org membership 表示テスト。検証: npm run test --workspace api -- src/routes/__tests__/collections.test.ts.

File: apps/api/src/routes/__tests__/collections.test.ts [hunk 2]
```diff
@@ -461,6 +872,289 @@ describe("collections routes", () => {
...
+            method: "GET",
+            headers: { Authorization: `Bearer ${await createTestJWT({ sub: "user-1" })}` },
+        }, env);
+        expect(res.status).toBe(200);
+        const data = await res.json();
+        expect(data.papers.length).toBe(2);
+        const ids = data.papers.map((p) => p.id);
+        expect(ids).toContain("p1");
+        expect(ids).toContain("p2");
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/collections.test.ts]: 対応しました: 消費されない4つ目の select mock を削除し、テストのレスポンス型も明示して typecheck が通るようにしました。影響範囲: collections route tests。検証: npm run typecheck --workspace api / npm run test --workspace api -- src/routes/__tests__/collections.test.ts.

File: apps/api/src/routes/__tests__/collections.test.ts [hunk 3]
```diff
@@ -461,6 +872,289 @@ describe("collections routes", () => {
...
+        const app = await createTestApp();
+        const env = createTestEnv();
+        const res = await app.request("http://localhost/api/collections/c1/papers", {
+            method: "PATCH",
+            headers: { Authorization: `Bearer ${await createTestJWT({ sub: "user-1" })}` },
+            body: JSON.stringify({ paper_ids: ["p1"] }),
+        }, env);
+        expect(res.status).toBe(400);
+        expect(await res.json()).toEqual({ error: "paper_ids must include all papers in collection" });
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/collections.test.ts]: 対応しました: 指摘の重複テストは現行差分では削除済みで、既存の requires every existing paper to be included テストに一本化されています。検証: npm run test --workspace api -- src/routes/__tests__/collections.test.ts.

File: apps/api/src/routes/__tests__/collections.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/collections.test.ts]: 対応しました: private paper のケースでは orgOnlyIds が空になり追加 select が発生しないため、未使用の4つ目の mock を削除しました。検証: npm run typecheck --workspace api / npm run test --workspace api -- src/routes/__tests__/collections.test.ts.

File: apps/api/src/routes/__tests__/collections.test.ts [hunk 4]
```diff
@@ -461,6 +872,289 @@ describe("collections routes", () => {
...
+        const app = await createTestApp();
+        const env = createTestEnv();
+        const res = await app.request("http://localhost/api/collections/c1/papers", {
+            method: "POST",
+            headers: { Authorization: `Bearer ${await createTestJWT({ sub: "user-1" })}` },
+            body: "{invalid-json}",
+        }, env);
+        expect(res.status).toBe(400);
+        expect(await res.json()).toEqual({ error: "Invalid JSON body" });
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/collections.test.ts]: 対応しました: 指摘の重複テストは現行差分では残っておらず、別経路の PATCH invalid JSON テストのみになっています。検証: npm run test --workspace api -- src/routes/__tests__/collections.test.ts.
